frame/society/src/benchmarking.rs

```diff
@@ -0,0 +1,384 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Society pallet benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{account, benchmarks_instance_pallet, whitelisted_caller};
+use frame_system::RawOrigin;
+
+use sp_runtime::traits::Bounded;
+
+use crate::Pallet as Society;
+
+// Set up Society
+fn setup_society<T: Config<I>, I: 'static>(founder: T::AccountId) -> Result<(), &'static str> {
+	let candidate_deposit = T::Currency::minimum_balance();
+	let origin = T::FounderSetOrigin::successful_origin();
+	let max_members = 5u32;
+	let max_intake = 3u32;
+	let max_strikes = 3u32;
+	Society::<T, I>::found_society(
+		origin,
+		founder,
+		max_members,
+		max_intake,
+		max_strikes,
+		candidate_deposit,
+		b"benchmarking-society".to_vec(),
+	)?;
+	Ok(())
+}
+
+fn add_candidate<T: Config<I>, I: 'static>(
+	name: &'static str,
+	tally: Tally,
+	skeptic_struck: bool,
+) -> T::AccountId {
+	let candidate: T::AccountId = account(name, 0, 0);
+	T::Currency::make_free_balance_be(&candidate, BalanceOf::<T, I>::max_value());
+	let deposit: BalanceOf<T, I> = T::Currency::minimum_balance();
+
+	let candidacy = Candidacy {
+		round: RoundCount::<T, I>::get(),
+		kind: BidKind::Deposit(deposit),
+		bid: 0u32.into(),
+		tally,
+		skeptic_struck,
+	};
+	Candidates::<T, I>::insert(&candidate, &candidacy);
+	candidate
+}
+
+fn increment_round<T: Config<I>, I: 'static>() {
+	let mut round_count = RoundCount::<T, I>::get();
+	round_count.saturating_inc();
+	RoundCount::<T, I>::put(round_count);
+}
+
+benchmarks_instance_pallet! {
+
+	bid {
+		let founder: T::AccountId = account("founder", 0, 0);
+		setup_society::<T, I>(founder)?;
+
+		let caller: T::AccountId = whitelisted_caller();
+		let deposit: BalanceOf<T, I> = T::Currency::minimum_balance();
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T, I>::max_value());
+	}: _(RawOrigin::Signed(caller.clone()), 10u32.into())
```

Comment:
You should bit with a multiple of `minimum_balance`. This value is smaller than the existential deposit in all real chains and might have unforeseen consequences.